otherlibs/unix/gethost.c

```diff
@@ -84,28 +85,42 @@ CAMLprim value caml_unix_gethostbyaddr(value a)
 {
   struct in_addr adr = GET_INET_ADDR(a);
   struct hostent * hp;
+  int addr_type;
+  socklen_t addr_len;
+#if HAS_IPV6
+  if (caml_string_length(a) == 16) {
+    addr_type = AF_INET6;
+    addr_len = 16;
+  }
+  else {
+    addr_type = AF_INET;
+    addr_len = 4;
+  }
+#else
+  addr_type = AF_INET;
```

Comment:
`addr_len` needs to be set to 4 in this case, right? In fact, it may be cleaner to set `addr_type` to `AF_INET` and `addr_len` to `4` to begin with and modify it to `(AF_INET6, 16)` if IP6 is available and `a` has length 16.

Reply:
Indeed, thanks for noticing this.